src/generator/AutoRest.Ruby.Azure/Model/CompositeTypeRba.cs

```diff
@@ -42,7 +42,10 @@ public override string GetBaseTypeName()
                 if (this.BaseModelType.Extensions.ContainsKey(AzureExtensions.ExternalExtension) ||
                     this.BaseModelType.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension))
                 {
-                    typeName = "MsRestAzure::" + typeName;
+                    if (!typeName.Equals("SubResource"))
```

Comment:
Let me know what you think about @devigned comments on how to best interpret `Subresource` and what we settle on. Based on that this may change :).